rpcs3/Emu/Cell/Modules/cellCamera.cpp

```diff
@@ -280,14 +315,15 @@ error_code cellCameraInit()
 
 	std::lock_guard lock(g_camera.mutex);
 
-	if (g_camera.init)
+	const bool is_null = g_cfg.io.camera == camera_handler::null;
+
+	if (!g_camera.state.compare_and_swap_test(camera_state::uninit, is_null ? camera_state::detached : camera_state::closed))
```

Comment:
Why is it initialized here before the function is done?

Reply:
Why not?

Reply:
Less LOC.